packages/@aws-cdk/aws-appmesh/lib/private/utils.ts

```diff
@@ -81,3 +82,17 @@ export function renderListenerTlsOptions(scope: Construct, listenerTls: Listener
     }
     : undefined;
 }
+
+/**
+ * This is the helper method to check if mesh sharing is enabled (mesh owner field needs to be present)
+ */
+export function renderMeshOwner(resourceAccount: string, meshAccount: string) : string | undefined {
+  if ( !Token.isUnresolved(resourceAccount) && !Token.isUnresolved(meshAccount)) {
+    if (Token.compareStrings(resourceAccount, meshAccount) === TokenComparison.DIFFERENT) {
+      // Mesh sharing is enabled, populate mesh owner
+      return meshAccount;
+    }
+  }
+  // Don't populate mesh owner
+  return undefined;
```

Comment:
You don't need the `.isUnresolved()` check - `Token.compareStrings()` does this for you already. Just do:

```suggestion
  const comparison = Token.compareStrings(resourceAccount, meshAccount);
  return comparison === TokenComparison.DIFFERENT || comparison === TokenComparison.ONE_UNRESOLVED
    ? meshAccount
    : undefined;
```